bionmr_utils/md/process/select.py

```diff
@@ -19,67 +31,60 @@ def get_methyl_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
                 ('ILE', 'CG2'): ['CG2', 'HG21'],
                 ('MET', 'CE1'): ['CE1', 'HE1']}
 
-    atom_pairs = []
-
-    for r in frame.asResidues:
-        for atom in r.asAtoms:
-            if (r.name.str, atom.name.str) in CH3_dict.keys():
-                C = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][0])]
-                H = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][1])]
-                atom_pairs.append((C, H))
-
-    return atom_pairs
+    C_ids = []
+    H_ids = []
 
+    for r in frame.residues:
+        for atom in r.atoms:
+            if (r.name, atom.name) in CH3_dict.keys():
+                C_ids.append(r[CH3_dict[(r.name, atom.name)][0]].id)
+                H_ids.append(r[CH3_dict[(r.name, atom.name)][1]].id)
 
-def get_NH_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
-    """
+    C_atoms = frame.atoms.filter(aId.is_in(set(C_ids)))
+    H_atoms = frame.atoms.filter(aId.is_in(set(H_ids)))
+    atom_pairs_selection = (C_atoms, H_atoms)
 
-    :param frame: Frame
-    :return: list of all backbone atom pairs of given frame.
-    """
-    atom_pairs = []
+    return atom_pairs_selection
 
-    for r in frame.asResidues:
-        try:
-            atom_pairs.append((r[AtomName("N")], r[AtomName("H")]))
-        except pyxmolpp2.polymer.OutOfRangeResidue:
-            pass
 
-    return atom_pairs
+class UnpairedElectronSelection:
+    def __init__(self, nitrogen_selection, oxygen_selection):
+        self.nitrogen_selection = nitrogen_selection
+        self.oxygen_selection = oxygen_selection
 
+    class ElectronCoords:
+        def __init__(self, nitrogen_coords, oxygen_coords):
+            self.nitrogen_coords = nitrogen_coords
+            self.oxygen_coords = oxygen_coords
 
-class UnpairedElectron:
-    def __init__(self, nitrogen, oxygen):
-        self.nitrogen = nitrogen
-        self.oxygen = oxygen
+        @property
+        def values(self):
+            return (self.nitrogen_coords.values + self.oxygen_coords.values) / 2
 
     @property
-    def rId(self):
-        return self.nitrogen.rId
+    def coords(self):
+        return self.ElectronCoords(self.nitrogen_selection.coords, self.oxygen_selection.coords)
 
     @property
-    def aName(self):
-        return AtomName("e")
+    def size(self):
+        return self.nitrogen_selection.size
 
     @property
-    def r(self):
-        return (self.nitrogen.r + self.oxygen.r) / 2
+    def index(self):
+        return self.nitrogen_selection.index
 
 
-def get_mtsl_vectors(frame: Frame) -> List[Tuple[Atom, UnpairedElectron]]:
+def get_mtsl_selection(frame: Frame,
+                       label_name="CML",
+                       ) -> Tuple[AtomSelection, UnpairedElectronSelection]:

```

Comment:
Return type annotation is not consistent with returned type (pair is swapped). In this case I think we can avoid explicit annotation since it can be easily deduced from implementation. 
```suggestion
                       ):
```